onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -957,6 +1089,402 @@ Status TreeEnsembleCommonClassifier<InputType, ThresholdType, OutputType>::compu
+    const std::vector<ThresholdType>& membership_values, std::vector<std::vector<ThresholdType>>& membership_values_by_id,
+    const std::vector<uint8_t>& nodes_modes) {
+  membership_values_by_id.clear();
+  membership_values_by_id.reserve(nodes_modes.size());
+
+  size_t curr_id = 0;
+  for (const auto node_mode : nodes_modes) {
+    membership_values_by_id.emplace_back();
+    if (node_mode != static_cast<int64_t>(NODE_MODE_V5::BRANCH_MEMBER)) {
+      continue;
+    }
+
+    while (curr_id < membership_values.size() && !_isnan_(membership_values[curr_id])) {
+      membership_values_by_id.back().push_back(membership_values[curr_id++]);
+    }
+    curr_id++;
+  }
+}
+
+template <typename IOType, typename ThresholdType>
+int64_t TreeEnsembleCommonV5<IOType, ThresholdType>::transformInputOneTree(
+    const size_t curr_id, const int64_t curr_treeid, const int64_t curr_nodeid, const size_t curr_membership_value_id, const bool is_leaf,
+    // input
+    const std::vector<int64_t>& leaf_targetids, const std::vector<ThresholdType>& leaf_weights, const std::vector<std::vector<ThresholdType>>& membership_values_by_id,
+    const std::vector<int64_t>& nodes_falseleafs, const std::vector<int64_t>& nodes_falsenodeids, const std::vector<int64_t>& nodes_featureids,
+    const std::vector<ThresholdType>& nodes_hitrates, const std::vector<int64_t>& nodes_missing_value_tracks_true, const std::vector<uint8_t>& nodes_modes,
+    const std::vector<ThresholdType>& nodes_splits, const std::vector<int64_t>& nodes_trueleafs, const std::vector<int64_t>& nodes_truenodeids,
+    const std::vector<int64_t>& tree_roots,
+    // output
+    std::vector<int64_t>& nodes_falsenodeids_old, std::vector<int64_t>& nodes_featureids_old, std::vector<ThresholdType>& nodes_hitrates_as_tensor_old,
+    std::vector<int64_t>& nodes_missing_value_tracks_true_old, std::vector<uint8_t>& nodes_modes_old, std::vector<int64_t>& nodes_nodeids_old,
+    std::vector<int64_t>& nodes_treeids_old, std::vector<int64_t>& nodes_truenodeids_old, std::vector<ThresholdType>& nodes_values_as_tensor_old,
+    std::vector<int64_t>& target_class_ids_old, std::vector<int64_t>& target_class_nodeids_old, std::vector<int64_t>& target_class_treeids_old,
+    std::vector<ThresholdType>& target_class_weights_as_tensor_old) {
+  nodes_nodeids_old.push_back(curr_nodeid);
+  nodes_treeids_old.push_back(curr_treeid);
+
+  if (is_leaf) {
+    nodes_modes_old.push_back(static_cast<uint8_t>(NODE_MODE_V5::LEAF));
+    target_class_ids_old.push_back(leaf_targetids[curr_id]);
+    target_class_nodeids_old.push_back(curr_nodeid);
+    target_class_treeids_old.push_back(curr_treeid);
+    target_class_weights_as_tensor_old.push_back(leaf_weights[curr_id]);
+
+    // the below are irrelevant for a `LEAF`
+    nodes_featureids_old.push_back(-1);
+    nodes_truenodeids_old.push_back(-1);
+    nodes_falsenodeids_old.push_back(-1);
+    nodes_values_as_tensor_old.push_back(-1);
```

Comment:
```suggestion
    nodes_featureids_old.push_back(0);
    nodes_truenodeids_old.push_back(0);
    nodes_falsenodeids_old.push_back(0);
    nodes_values_as_tensor_old.push_back(0);
```

Even for leaves, these values can be used as an index in `ProcessTreeNodeLeave`, so they shouldn't be negative.